DFC.ServiceTaxonomy.CompUi/Repository/DataRepository.cs

```diff
@@ -49,6 +47,57 @@ public async Task<IEnumerable<RelatedContentData>> GetRelatedContentDataByConten
             return resultList;
         }
 
+        /// <summary>
+        /// Get the related content Ids for a particular item, where there is no link from the actual item back to the item being processed. 
+        /// </summary>
+        /// <param name="processing"></param>
+        /// <returns>IEnumerable<RelatedContentData></returns>
+        public async Task<IEnumerable<RelatedContentData>> GetRelatedContentDataByContentItemId(Processing processing)
+        {
+            IEnumerable<RelatedContentData>? resultList = new List<RelatedContentData>();
+
+            try
+            {
+                var sql = $"SELECT RelatedContentIds FROM RelatedContentItemIndex WITH (NOLOCK) " +
+                    $"WHERE ContentItemId = @contentItemId ";
+                var relatedContentIds = await ExecuteQuery<string>(sql, new { contentItemId = processing.ContentItemId });
+
+                if (relatedContentIds != null)
+                {
+                    var items = relatedContentIds?.FirstOrDefault()?.ToString().Split(',');
+                    var documentIdList = new List<int>();
+                    foreach (var itemId in items)
+                    {
+                        sql = $"SELECT DocumentId FROM RelatedContentItemIndex WITH (NOLOCK) " +
+                          $"WHERE ContentItemId = @contentItemId " +
+                          $"AND ContentType = 'JobProfile' ";
+                        var documentIds = await ExecuteQuery<int>(sql, new { contentItemId =  itemId.Replace("'", "") });
+
+                        documentIdList.AddRange(documentIds);
+                    }
```

Comment:
Again, my preference for LINQ comes in to play. Also, nothing wrong with what you have done, up to you if you want to implement the below.

```
sql = $"SELECT DocumentId FROM RelatedContentItemIndex WITH (NOLOCK) " +
                          $"WHERE ContentItemId = @contentItemId " +
                          $"AND ContentType = 'JobProfile' ";
var documentIdList = items
    .SelectMany(async x => await ExecuteQuery<int>(sql, new { contentItemId = x.Replace("'", "") }));
```

Reply:
Ahh yes. I think this may work, but not sure if it is better than what you already have.

```
var documentIdList =
                        (await Task.WhenAll(items.Select(x =>
                            ExecuteQuery<int>(sql, new { contentItemId = x.Replace("'", "") }))))
                        .SelectMany(x => x).ToList();
```

Again, up to you if you want to use the LINQ version or not :)

Reply:
Thanks @abo-olsen-DFE that works a treat.